File: src/main/java/br/com/leonardoferreira/jirareport/repository/LeadTimeConfigRepository.java
package br.com.leonardoferreira.jirareport.repository;

import java.util.List;

import br.com.leonardoferreira.jirareport.domain.LeadTimeConfig;
import java.util.Optional;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface LeadTimeConfigRepository extends CrudRepository<LeadTimeConfig, Long> {

    List<LeadTimeConfig> findByBoardId(Long boardId);

    Optional<LeadTimeConfig> findByIdAndBoardId(Long id, Long boardId);

}

File: database-library/src/main/kotlin/br/com/jiratorio/repository/LeadTimeConfigRepository.kt
package br.com.jiratorio.repository

import br.com.jiratorio.domain.entity.LeadTimeConfigEntity
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface LeadTimeConfigRepository : CrudRepository<LeadTimeConfigEntity, Long> {

    fun findByBoardId(boardId: Long): List<LeadTimeConfigEntity>

    fun findByIdAndBoardId(id: Long, boardId: Long): LeadTimeConfigEntity?

    fun findByIdOrNull(id: Long): LeadTimeConfigEntity? =
        findById(id).orElse(null)

}
